lightclient/src/client.rs

```diff
@@ -11,14 +13,38 @@ use tokio::sync::{mpsc, mpsc::error::SendError, oneshot};
 
 use super::platform::build_platform;
 
-pub const LOG_TARGET: &str = "light-client";
+pub const LOG_TARGET: &str = "subxt-light-client";
+
+/// A raw light-client RPC implementation that can connect to multiple chains.
+#[derive(Clone)]
+pub struct RawLightClientRpc {
+    /// Communicate with the backend task that multiplexes the responses
+    /// back to the frontend.
+    to_backend: mpsc::UnboundedSender<FromSubxt>,
+}
+
+impl RawLightClientRpc {
+    /// Construct a [`LightClientRpc`] that can communicated with the provided chain.
+    ///
+    /// # Note
+    ///
+    /// This uses the same underlying instance created by [`LightClientRpc::new_from_client`].
+    pub fn for_chain(&self, chain_id: smoldot_light::ChainId) -> LightClientRpc {
```

Comment:
I found the naming of this a bit odd,  

I would prefer `connect_to_chain(id: ChainId)`

Reply:
Another question is it possible to connect the same chain more than once?

Reply:
Yep, its possible to use the same chain ID to create multiple instances of the light client

Reply:
I guess I like `for_chain` better than `connect_to_chain` because I think it's already connected, so all we are doing I believe is "selecting" a chain from those that are available to work with. (maybe `seelct_chain` would also work for that reason; I'm easy either way)